lib/spack/env/cc

```diff
@@ -124,16 +135,30 @@ prepend() {
 # assuming the list is separated by SEP.
 # Handles empty lists and single-element lists.
 append() {
-    varname="$1"
-    elt="$2"
-
-    if empty "$varname"; then
-        eval "$varname=\"\${elt}\""
+    if [ -z "$3" ]; then
+        setsep "$1"
     else
-        # Get the appropriate separator for the list we're appending to.
-        setsep "$varname"
-        eval "$varname=\"\${$varname}${sep}\${elt}\""
+        sep="$3"
     fi
+    eval "$1=\"\${$1:+\$$1${sep}}\${2}\""
+}
+
+# appends LISTNAME ELEMENT SEP
+#
+# Append ELEMENT to the list stored in the variable LISTNAME,
+# assuming the list is separated by SEP.
+# Handles empty lists and single-element lists.
+appends() {
+    eval "$1=\"\${$1:+\$$1${3}}\${2}\""
+}
```

Comment:
This appears not to bee used anywhere

Reply:
That appears to be the case.  I used to have appendsv built on top of it, but no longer. If you strongly prefer I can remove it, but this version is cheaper than append and more convenient than appendsv when appending a value that is not currently stored in a variable.